src/Api/HL/Controller/ProjectController.php

```diff
@@ -95,6 +134,72 @@ protected static function getRawKnownSchemas(): array
             'Task' => [
                 'x-itemtype' => ProjectTask::class,
                 'type' => Doc\Schema::TYPE_OBJECT,
+                'x-rights-conditions' => [ // Object-level extra permissions
+                    'read' => static function () {
+                        if (!\Session::haveRight(Project::$rightname, Project::READALL)) {
+                            if (!\Session::haveRight(Project::$rightname, Project::READMY)) {
+                                return false; // Deny reading
+                            }
+                            $project_criteria = [
+                                'LEFT JOIN' => [
+                                    'glpi_projectteams' => [
+                                        'ON' => [
+                                            'glpi_projectteams' => 'projects_id',
+                                            'project' => 'id'
+                                        ]
+                                    ]
+                                ],
+                                'WHERE' => [
+                                    'OR' => [
+                                        '_.users_id' => \Session::getLoginUserID(),
+                                        [
+                                            "glpi_projectteams.itemtype"   => 'User',
+                                            "glpi_projectteams.items_id"   => \Session::getLoginUserID()
+                                        ]
+                                    ]
+                                ]
+                            ];
+                            if (count($_SESSION['glpigroups'])) {
+                                $project_criteria['WHERE']['OR'][] = [
+                                    'project.groups_id' => $_SESSION['glpigroups'],
+                                ];
+                                $project_criteria['WHERE']['OR'][] = [
+                                    "glpi_projectteams.itemtype"   => 'Group',
+                                    "glpi_projectteams.items_id"   => $_SESSION['glpigroups']
+                                ];
+                            }
+
+                            $criteria = [
+                                'LEFT JOIN' => [
```

Comment:
I do not see where the `project` table is joined here.

Reply:
Automatically joined because of the presence of the `project` (refers to the parent project) property in the schema. Since it isn't currently possible for the user to remove properties from the result, this should always be joined.